controller/node_controller_test.go

```diff
@@ -621,233 +1017,349 @@ func (s *TestSuite) TestSyncNode(c *C) {
+					newNodeCondition(longhorn.NodeConditionTypeMountPropagation, longhorn.ConditionStatusTrue, ""),
 				},
+				DiskStatus: map[string]*longhorn.DiskStatus{},
 			},
-		}, false)
-	extraInstanceManager.Spec.Image = TestExtraInstanceManagerImage
-
-	tc.instanceManagers = map[string]*longhorn.InstanceManager{
-		TestInstanceManagerName: newInstanceManager(
-			TestInstanceManagerName, longhorn.InstanceManagerStateRunning,
-			TestOwnerID1, TestNode1, TestIP1,
-			map[string]longhorn.InstanceProcess{},
-			map[string]longhorn.InstanceProcess{},
-			false,
-		),
+		},
+		instanceManagers: map[string]*longhorn.InstanceManager{
+			TestInstanceManagerName: newInstanceManager(
+				TestInstanceManagerName, longhorn.InstanceManagerStateRunning,
+				TestOwnerID1, TestNode1, TestIP1,
+				map[string]longhorn.InstanceProcess{},
+				map[string]longhorn.InstanceProcess{},
+				false,
+			),
+		},
 	}
 
-	tc.expectInstanceManagers = map[string]*longhorn.InstanceManager{
-		TestInstanceManagerName: newInstanceManager(
-			TestInstanceManagerName, longhorn.InstanceManagerStateRunning,
-			TestOwnerID1, TestNode1, TestIP1,
-			map[string]longhorn.InstanceProcess{},
-			map[string]longhorn.InstanceProcess{},
-			false,
-		),
+	s.initTest(c, tc)
+
+	for _, node := range tc.lhNodes {
+		if s.controller.controllerID == node.Name {
+			err = s.controller.diskMonitor.RunOnce()
+			c.Assert(err, IsNil)
+		}
+
+		err = s.controller.syncNode(getKey(node, c))
+		c.Assert(err, IsNil)
+
+		n, err := s.lhClient.LonghornV1beta2().Nodes(TestNamespace).Get(context.TODO(), node.Name, metav1.GetOptions{})
+		c.Assert(err, IsNil)
+
+		s.checkNodeConditions(c, tc, expectation, n)
 	}
-	tc.expectOrphans = []*longhorn.Orphan{}
-	testCases["clean up all instance managers if there is no disk on the node"] = tc
 
-	for name, tc := range testCases {
-		fmt.Printf("testing %v\n", name)
+	s.checkInstanceManagers(c, tc, expectation)
+}
 
-		kubeClient := fake.NewSimpleClientset()
-		lhClient := lhfake.NewSimpleClientset()
+// -- Helpers --
 
-		informerFactories := util.NewInformerFactories(TestNamespace, kubeClient, lhClient, controller.NoResyncPeriodFunc())
+func (s *NodeControllerSuite) checkNodeConditions(c *C, tc *TestCase, exp *Expectation, node *longhorn.Node) {
+	// Check that all node status conditions match the expected node status
+	// conditions - save for the last transition timestamp and the actual
+	// message
+	for idx, condition := range node.Status.Conditions {
+		condition.LastTransitionTime = ""
+		condition.Message = ""
+		node.Status.Conditions[idx] = condition
+	}
+	c.Assert(node.Status.Conditions, DeepEquals, exp.nodeStatus[node.Name].Conditions)
+}
 
-		nIndexer := informerFactories.LhInformerFactory.Longhorn().V1beta2().Nodes().Informer().GetIndexer()
-		pIndexer := informerFactories.KubeInformerFactory.Core().V1().Pods().Informer().GetIndexer()
+func (s *NodeControllerSuite) checkDiskConditions(c *C, tc *TestCase, exp *Expectation, node *longhorn.Node) {
```

Comment:
Looks like we don't call this function in any the test cases. Is it expected?

Reply:
Good spot, this needs to be called for all tests that have non-empty disk conditions in their expectations.